### PR TITLE
WAI-41 Average Cohort Survival

### PR DESCRIPTION
**Pull request by EMcQ-BES**
### Issue #:
Resolves WAI-41
NZ-41

### Changes:

- Add an attribute type to non-target districts
- Add a new HLO1 list visual for primary education at country level
- Add indicators to calculate average vs target cohort survival rate
- Add a drilldown graph

---

### Screenshots:
See attached ticket

### CODE REVIEW

**Review by edmofro — CHANGES_REQUESTED** [review]
Before I review the indicators etc., could you please update the report to use the [new config](https://beyond-essential.slab.com/posts/report-server-transform-config-design-change-proposals-k8l3tch0) that just got merged to dev? Cheers!

File: packages/database/src/migrations/20210907222655-LESMISPrimaryEducationCohortSurvivalRate-modifies-data.js [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+
+exports.up = async function (db) {
+  await generateAllIndicators(db);
+  for (const { code, frontEndConfig, reportConfig } of DASHBOARD_ITEMS) {
+    await addNewDashboardItemAndReport(db, {
+      code,
+      frontEndConfig,
+      reportConfig,
+      permissionGroup: 'LESMIS Public',
+      dashboardCode: 'LESMIS_ESSDP_PrimarySubSector_HLO1',
```
Comment (EMcQ-BES) [inline packages/database/src/migrations/20210907222655-LESMISPrimaryEducationCohortSurvivalRate-modifies-data.js:407]: Just realised that I only actually added this dashboard in WAI-44 so while it was working on my local it actually relies on [another PR](https://github.com/beyondessential/tupaia/pull/3171)

I'm just going to wait until that PR gets merged so I don't have to do complicated git stuff

File: packages/database/src/migrations/20210907222655-LESMISPrimaryEducationCohortSurvivalRate-modifies-data.js [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+const generateFiveYearIndicator = (code, dataElement, target) => ({
+  id: generateId(),
+  code,
+  builder: 'analyticArithmetic',
+  config: {
+    formula: `${dataElement} >= ${target} ? 1 : ${dataElement} >= firstExistingValue(value1YearAgo, value2YearsAgo, value3YearsAgo, value4YearsAgo, value5YearsAgo, Infinity) ? 0 : -1`,
+    parameters: [
+      {
+        builder: 'analyticArithmetic',
+        // 1 year is defined separately because I don't want to handle singular/plural year(s)
```
Comment (edmofro) [inline packages/database/src/migrations/20210907222655-LESMISPrimaryEducationCohortSurvivalRate-modifies-data.js:98]: And fair enough too

File: packages/database/src/migrations/20210907222655-LESMISPrimaryEducationCohortSurvivalRate-modifies-data.js [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+      columns: {
+        "=translate($dataElement, { csr_priority_district_average: 'Priority Districts Average', csr_non_priority_district_average: 'Other Districts Average',  csr_priority_district_target: 'Priority Districts Target', csr_non_priority_district_target: 'Other Districts Target' })":
+          '=$value/100',
+        timestamp: '=periodToTimestamp($period)',
+      },
+    },
+    {
+      transform: 'mergeRows',
+      groupBy: 'timestamp',
+      using: 'last',
```
Comment (edmofro) [inline packages/database/src/migrations/20210907222655-LESMISPrimaryEducationCohortSurvivalRate-modifies-data.js:169]:
```suggestion
```
Default is now `single`, which adds the extra assertion that we are only expecting one to be defined

File: packages/database/src/migrations/20210907222655-LESMISPrimaryEducationCohortSurvivalRate-modifies-data.js [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+        type: 'MOST_RECENT',
+      },
+    ],
+  },
+  transform: [
+    'keyValueByDataElementName',
+    {
+      transform: 'mergeRows',
+      groupBy: 'organisationUnit',
+      using: 'last',
```
Comment (edmofro) [inline packages/database/src/migrations/20210907222655-LESMISPrimaryEducationCohortSurvivalRate-modifies-data.js:212]: Same deal

**Review by edmofro — APPROVED** [review]
Couple of suggestions but lgtm!

**Review by edmofro — APPROVED** [review]

**Pull request merged**